EShop.Web/Services/InMemoryRepository.cs

```diff
@@ -5,18 +5,62 @@
 
 namespace EShop.Web.Services
 {
-    public class InMemoryRepository<T> : IRepository<T>
+    public class InMemoryRepository<T> : IRepository<T> where T : IIdentifiable
     {
         private readonly List<T> entities = new List<T>();
 
+        public int lastId = 0;
+
         public void Add(T entity)
         {
+            this.lastId = this.lastId + 1;
+            entity.Id = this.lastId;
```

Comment:
`entity.Id = ++this.lastId;` ar fi echivalent la aceste 2 lini

`++variabila => (variabila = variabila + 1)`
rezultatul e intors ca si valoare, drept urmare, poti sa o pasezi direct peste entity.Id